File: 백준/Silver/10866. 덱/덱.cc
#include <iostream>
#include <deque>
#include <string>

using namespace std;

void command(deque<int> &dq, string &str) {
    if (str == "push_back") {
        int x;
        cin >> x;
        dq.push_back(x);
    }
    
    else if (str == "push_front") {
        int x;
        cin >> x;
        dq.push_front(x);
    }
    
    else if (str == "front") {
        if(!dq.empty()) cout << dq.front() << "\n";
        else cout << -1 << "\n";
    }
    
    else if (str == "back") {
        if(!dq.empty()) cout << dq.back() << "\n";
        else cout << -1 << "\n";
    }
    
    else if (str == "size") {
        cout << dq.size() << "\n";
    }
    
    else if (str == "empty") {
        cout << dq.empty() << "\n";
    }
    
    else if (str == "pop_front") {
        if(!dq.empty()) {
            cout << dq.front() << "\n";
            dq.pop_front();
        } 
        else cout << -1 << "\n";
    }
    
    else if (str == "pop_back") {
        if(!dq.empty()) {
            cout << dq.back() << "\n";
            dq.pop_back();
        } 
        else cout << -1 << "\n";
    }
    
    else;
}

int main(int argc, char** argv) {
    int N;
    deque<int> dq;
    cin >> N;
    
    for(auto i = 0; i < N; i++) {
        string str;
        cin >> str;
        command(dq, str);
    }
    
    return 0;
}

File: 백준/Silver/10866. 덱/덱.py
import sys
from collections import deque

dq = deque()

# Reading the number of commands
number = int(sys.stdin.readline().strip())

def check_empty(dq):
    return not len(dq)

for _ in range(number):
    command = sys.stdin.readline().split()
    
    if command[0] == "push_front":
        dq.appendleft(int(command[1]))
        
    elif command[0] == "push_back":
        dq.append(int(command[1]))
        
    elif command[0] == "pop_front":
        if check_empty(dq):
            print(-1)
        else:
            print(dq.popleft())
    
    elif command[0] == "pop_back":
        if check_empty(dq):
            print(-1)
        else:
            print(dq.pop())
    
    elif command[0] == "size":
        print(len(dq))
    
    elif command[0] == "empty":
        print(1 if check_empty(dq) else 0)
    
    elif command[0] == "front":
        print(dq[0] if not check_empty(dq) else -1)
            
    elif command[0] == "back":
        print(dq[-1] if not check_empty(dq) else -1)